From 258db61bd8409e3c2f93f2eac16ffed2bfc221df Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Sun, 18 Jul 2021 01:03:54 +0200
Subject: [PATCH] Allow None buildtype

Some distributions, in my case Alpine Linux, require building CMake packages with the
"None" buildtype as this makes packages use compiler flags required by
the distribution.

See a discussion with the Arch Linux maintainer:
archlinux/svntogit-community@da37cf2#r53047679

And with one of the Alpine Linux packagers:
https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/23363#note_168960
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b924921e41..1a85dfb9a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,8 +53,8 @@ if(NOT CMAKE_CONFIGURATION_TYPES)
     set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
     # Set the possible values of build type for cmake-gui
     set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "RelWithDebInfo")
-  elseif(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo)$")
-    message(FATAL_ERROR "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} is not supported, use one of Debug, Release or RelWithDebInfo.")
+  elseif(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|None)$")
+    message(FATAL_ERROR "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} is not supported, use one of Debug, Release, RelWithDebInfo or None.")
   endif()
 endif()
 
-- 
2.32.0

